### DISCUSSION TITLE
lotus Cannot start after forced shutdown

### DISCUSSION THREAD

**Discussion by yzn5555** [filecoin-project/lotus#7951] [Build and Sync]
Execute the kill -9 command to close the program, start again and report an error

ERROR: initializing node: starting node: could not build arguments for function "github.com/filecoin-project/lotus/node/modules/lp2p".StartListening.func1 (/data/lotus/node/modules/lp2p/addrs.go:98): failed to build host.Host: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): failed to build lp2p.RawHost: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): could not build value group []config.Option[group="libp2p"]: could not build arguments for function "github.com/filecoin-project/lotus/node/modules/lp2p".ConnGaterOption (/data/lotus/node/modules/lp2p/conngater.go:14): failed to build *conngater.BasicConnectionGater: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): failed to build dtypes.MetadataDS: received non-nil error from function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): opening datastore /metadata: leveldb/journal: block/chunk corrupted: chunk length overflows block (134 bytes) [file=001704.log]

**Comment by yzn5555** [comment]
The damaged block is found. The data after the height of 279862 is damaged. How to repair it

**Comment by yzn5555** [comment]
Put the file 001704.log Move files to other directories，Start lotus program；The program can be started, but there are errors in the log，Check status:
![image](https://user-images.githubusercontent.com/36029121/100566985-673e0100-3302-11eb-8b64-870c9ee7f60f.png)
![image](https://user-images.githubusercontent.com/36029121/100567072-9e141700-3302-11eb-8f6e-d9323d6172c5.png)

**Comment by yzn5555** [comment]
I wonder if there is a command to repair the damaged block

**Comment by kernelogic** [comment]
Can you re-import the snapshot?

**Comment by yzn5555** [comment]
Now it has been restored, copy the data of other lotus node chain directory
I want to ask, if you import through a snapshot, does the snapshot do not contain node data?





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "filecoin-project/lotus"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年12月3日(星期四) 凌晨2:49
收件人:&nbsp;"filecoin-project/lotus"<lotus@noreply.github.com&gt;;
抄送:&nbsp;"1076813060"<1076813060@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [filecoin-project/lotus] lotus Cannot start after forced shutdown (#5051)





 
Can you re-import the snapshot?

**Comment by falcolee** [comment]
Same here.i run a local dev network.after shutdown lotus daemon with ctrl+c . it couldn't start up again with the error below:

ERROR: initializing node: starting node: could not build arguments for function "github.com/filecoin-project/lotus/node/modules/lp2p".StartListening.func1 (/home/worker/lotus/node/modules/lp2p/addrs.go:98): failed to build host.Host: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): failed to build lp2p.BaseIpfsRouting: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): failed to build dtypes.NetworkName: could not build arguments for function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): failed to build dtypes.AfterGenesisSet: received non-nil error from function "reflect".makeFuncStub (/usr/local/go/src/reflect/asm_amd64.s:12): genesis in the repo is not the one expected by this version of Lotus!

plus my command here:
./lotus daemon --lotus-make-genesis=devgen.car --genesis-template=localnet.json --bootstrap=false

**Reply by SgtCoin** [reply]
Not a full answer as I am sure that there are other ways this can come up, but here is a starting point. 

https://sgtcoin.github.io/everest-web/articles/genesis-in-repo/

**Reply by lqf1215** [reply]
Link failure error 404 here

**Comment by Reiers** [comment]
Hi @yzn5555 

Thanks for the report. The issue you are experiencing lies elsewhere.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing Lotus issue ticket.

**Comment by SgtCoin** [comment]
@yzn5555 

Hi, I am Sgt_Coin (Trey), on of the support engineers here. The snapshot import contains chain data, no host data. The reason for using the snapshot is to limit the amount of time to sync with the blockchain. You also shouldn't use `kill -9` to stop your lotus node processes. Instead, use `lotus daemon stop`. Please let me know if you are still facing any issues. I will be happy to help.

**Comment by SgtCoin** [comment]
Archiving for future reference as no response has been received.
